[test-triage] chip_sw_uart_smoketest_signed

### Hierarchy of regression failure

Chip Level

### Failure Description

  UVM_ERROR @ 20013.398695 us: (chip_sw_base_vseq.sv:218) uvm_test_top.env.virtual_sequencer [uvm_test_top.env.virtual_sequencer.chip_sw_base_vseq] SW TEST TIMED OUT. STATE: SwTestStatusBooted, TIMEOUT = 20000000 ns

### Steps to Reproduce

GH revision: [9c0b24ddb](https://github.com/lowrisc/opentitan/tree/9c0b24ddbeb5616bd8db661ec1395f2d4f326bc4)

util/dvsim/dvsim.py hw/top_earlgrey/dv/chip_sim_cfg.hjson -i chip_sw_uart_smoketest_signed --build-seed 2483310614 --fixed-seed 4088507876


### Tests with similar or related failures

_No response_

Note for devs. Looks like the timeout has gone but now have some bazel / fastbuild issues.

Since https://reports.opentitan.org/hw/top_earlgrey/dv/2022.08.16_16.23.56/results.html
chip_sw_uart_smoketest is passing but 
chip_sw_uart_smoketest_signed is still not passing.

Error observed between 12 and 16 August
```
fastbuild/bin/sw/device/tests/uart_smoketest_prog_sim_dv.test_key_0.signed.elf': No such file or directory
  make: *** [/workspace/mnt/repo_top/hw/dv/tools/dvsim/sim.mk:73: sw_build] Error 1
```

So now it may be a new type of issue
```
UVM_FATAL @   0.965959 us: (mem_bkdr_util.sv:480) [mem_bkdr_util[Rom]] file rom_sim_dv.39.scr.vmem could not be opened for r mode
```

--- a/hw/top_earlgrey/dv/chip_rom_tests.hjson
+++ b/hw/top_earlgrey/dv/chip_rom_tests.hjson
@@ -20,7 +20,7 @@
     {
       name: chip_sw_uart_smoketest_signed
       uvm_test_seq: chip_sw_base_vseq
-      sw_images: ["//sw/device/tests:uart_smoketest:1:signed"]
+      sw_images: ["//sw/device/tests:uart_smoketest_signed:1:signed"]
       en_run_modes: ["sw_test_mode_rom"]
       run_opts: ["+sw_test_timeout_ns=20000000"]
     }

--- a/rules/opentitan_test.bzl
+++ b/rules/opentitan_test.bzl
@@ -5,8 +5,17 @@
 load("@//rules:opentitan.bzl", "opentitan_flash_binary", "opentitan_rom_binary")
 load("@bazel_skylib//lib:shell.bzl", "shell")
 
-_EXIT_SUCCESS = r"PASS.*\n"
-_EXIT_FAILURE = r"(FAIL|FAULT).*\n"
+OTTF_SUCCESS_MSG = r"PASS.*\n"
+OTTF_FAILURE_MSG = r"(FAIL|FAULT).*\n"
+ROM_BOOT_FAILURE_MSG = "BFV:[0-9a-f]{8}"
+
+# These are defined for positive test cases and should be flipped for negative
+# test cases, i.e., when a test failure is the expected outcome.
+DEFAULT_TEST_SUCCESS_MSG = OTTF_SUCCESS_MSG
+DEFAULT_TEST_FAILURE_MSG = "({})|({})".format(
+    OTTF_FAILURE_MSG,
+    ROM_BOOT_FAILURE_MSG,
+)
 
 _BASE_PARAMS = {
     "args": [],
@@ -17,8 +26,8 @@ _BASE_PARAMS = {
     "tags": [],
     "test_runner": "@//util:opentitan_functest_runner.sh",
     "timeout": "moderate",  # 5 minutes
-    "exit_success": _EXIT_SUCCESS,
-    "exit_failure": _EXIT_FAILURE,
+    "exit_success": DEFAULT_TEST_SUCCESS_MSG,
+    "exit_failure": DEFAULT_TEST_FAILURE_MSG,
 }
 
 def dv_params(

--- a/sw/device/silicon_creator/rom/e2e/BUILD
+++ b/sw/device/silicon_creator/rom/e2e/BUILD
@@ -4,6 +4,8 @@
 
 load(
     "//rules:opentitan_test.bzl",
+    "DEFAULT_TEST_FAILURE_MSG",
+    "DEFAULT_TEST_SUCCESS_MSG",
     "cw310_params",
     "opentitan_functest",
     "verilator_params",
@@ -11,23 +13,15 @@ load(
 
 package(default_visibility = ["//visibility:public"])
 
-BOOT_FAILURE_MSG = "BFV:[0-9a-f]{8}"
-
-BOOT_SUCCESS_MSG = "PASS!"
-
 opentitan_functest(
     name = "e2e_bootup_success",
     srcs = ["empty_test.c"],
     cw310 = cw310_params(
         bitstream = "//hw/bitstream:rom",
-        exit_failure = BOOT_FAILURE_MSG,
-        exit_success = BOOT_SUCCESS_MSG,
     ),
     manifest = "//sw/device/silicon_creator/rom_ext:manifest_standard",
     signed = True,
     verilator = verilator_params(
-        exit_failure = BOOT_FAILURE_MSG,
-        exit_success = BOOT_SUCCESS_MSG,
         rom = "//sw/device/silicon_creator/rom",
         # Note: Leaving as broken since this test takes a long time and
         # we don't really need it as long as the CI runs the FPGA version.
@@ -47,8 +41,6 @@ opentitan_functest(
     srcs = ["empty_test.c"],
     cw310 = cw310_params(
         bitstream = "//hw/bitstream:rom_otp_dev",
-        exit_failure = BOOT_FAILURE_MSG,
-        exit_success = BOOT_SUCCESS_MSG,
         # TODO(lowRISC/opentitan#13603): Remove this "manual" tag when the
         # bitstream target can fetch pre-spliced bitstream from GCP.
         tags = ["manual"],
@@ -66,13 +58,13 @@ opentitan_functest(
     srcs = ["empty_test.c"],
     cw310 = cw310_params(
         bitstream = "//hw/bitstream:rom",
-        exit_failure = BOOT_SUCCESS_MSG,
-        exit_success = BOOT_FAILURE_MSG,
+        exit_failure = DEFAULT_TEST_SUCCESS_MSG,
+        exit_success = DEFAULT_TEST_FAILURE_MSG,
     ),
     signed = False,
     verilator = verilator_params(
-        exit_failure = BOOT_SUCCESS_MSG,
-        exit_success = BOOT_FAILURE_MSG,
+        exit_failure = DEFAULT_TEST_SUCCESS_MSG,
+        exit_success = DEFAULT_TEST_FAILURE_MSG,
         rom = "//sw/device/silicon_creator/rom",
     ),
     deps = [

--- a/sw/device/tests/BUILD
+++ b/sw/device/tests/BUILD
@@ -3,7 +3,14 @@
 # SPDX-License-Identifier: Apache-2.0
 
 load("//rules:opentitan.bzl", "OPENTITAN_CPU")
-load("//rules:opentitan_test.bzl", "cw310_params", "opentitan_functest", "verilator_params")
+load(
+    "//rules:opentitan_test.bzl",
+    "ROM_BOOT_FAILURE_MSG",
+    "cw310_params",
+    "dv_params",
+    "opentitan_functest",
+    "verilator_params",
+)
 
 # TODO(lowRISC:opentitan#13180): this is a temporary solution to enable writing
 # manufacturer specific tests in the `manufacturer_test_hooks` repository that
@@ -1345,10 +1352,35 @@ opentitan_functest(
 )
 
 opentitan_functest(
-    name = "uart_smoketest",
+    name = "uart_smoketest_signed",
     srcs = ["uart_smoketest.c"],
+    cw310 = cw310_params(
+        bitstream = "//hw/bitstream:rom",
+        exit_failure = ROM_BOOT_FAILURE_MSG,
+    ),
+    dv = dv_params(
+        rom = "//sw/device/silicon_creator/rom",
+    ),
     manifest = "//sw/device/silicon_creator/rom_ext:manifest_standard",
     signed = True,
+    verilator = verilator_params(
+        timeout = "eternal",
+        exit_failure = ROM_BOOT_FAILURE_MSG,
+        rom = "//sw/device/silicon_creator/rom",
+    ),
+    deps = [
+        "//hw/top_earlgrey/sw/autogen:top_earlgrey",
+        "//sw/device/lib/arch:device",
+        "//sw/device/lib/base:mmio",
+        "//sw/device/lib/dif:uart",
+        "//sw/device/lib/runtime:hart",
+        "//sw/device/lib/testing/test_framework:ottf_main",
+    ],
+)
+
+opentitan_functest(
+    name = "uart_smoketest",
+    srcs = ["uart_smoketest.c"],
     deps = [
         "//hw/top_earlgrey/sw/autogen:top_earlgrey",
         "//sw/device/lib/arch:device",

